src/NuGet.Core/NuGet.Build.Tasks/StaticGraphRestoreArguments.cs

```diff
@@ -0,0 +1,215 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Newtonsoft.Json;
+
+namespace NuGet.Build.Tasks
+{
+    /// <summary>
+    /// Represents arguments to the out-of-proc static graph-based restore which can be written to disk by <see cref="RestoreTaskEx" /> and then read by NuGet.Build.Tasks.Console.
+    /// </summary>
+    internal class StaticGraphRestoreArguments
+    {
+        /// <summary>
+        /// Gets or sets the path to the entry project.
+        /// </summary>
+        public string EntryProjectFilePath { get; set; }
+
+        /// <summary>
+        /// Gets or sets a <see cref="IEnumerable{T}" /> containing <see cref="KeyValuePair{TKey, TValue}" /> representing the global properties.
+        /// </summary>
+        public IEnumerable<KeyValuePair<string, string>> GlobalProperties { get; set; }
```

Comment:
```suggestion
        public IDictionary<string, string> GlobalProperties { get; set; }
```
or `IList`. Basically, anything other than `IEnumerable`.  I'm fine with `IEnumerable` being used as method return values, where it's expected to be used in chained LINQ queries. Personally, I think all other usage is an anti-pattern, and has caused large memory allocations, and therefore GC pauses, in our VS code, since every `foreach` on this (including LINQ queries) will cause an allocation on `.GetEnumerator()`.

Reply:
I did this so that the method that retrieves the global properties could have `yield return` so as to not make a copy of the global property dictionary.  The code that ends up using it is only enumerating it once and writing it to the arguments file.  Would you rather me refactor that method to return an actual `Dictionary<TKey, TValue>` or should I call `ToDictionary()` when I call it?

Reply:
I've just created a new dictionary and passed it around